agent/xds/listeners.go

```diff
@@ -1030,13 +1046,22 @@ func createDownstreamTransportSocketForConnectTLS(cfgSnap *proxycfg.ConfigSnapsh
 		return nil, fmt.Errorf("cannot inject peering trust bundles for kind %q", cfgSnap.Kind)
 	}
 
+	// Determine listener protocol type from configured service protocol. Don't hard fail on a config typo,
+	//The parse func returns default config if there is an error, so it's safe to continue.
+	cfg, _ := ParseProxyConfig(cfgSnap.Proxy.Config)
+
```

Comment:
🤔 Right now calling getAlpnProtocols with `tcp` is returning an implicit nil because it's not matched. To confirm, we want to add an case statement for tcp but it should still return the empty `alpnProtocols` slice.